---
title: "Genome-Wide Association Studies (GWAS) - Soybean response to off-target dicamba"
author: "Igor Kuivjogi Fernandes"
date: "`r Sys.Date()`"
bibliography: references.bib
output: pdf_document
---


# Introduction
The primary emphasis in both plant and animal studies has been on investigating the genetic origins of phenotypic variation, with broad goals of uncovering disease causes, enhancing agricultural practices, and comprehending adaptive mechanisms. In the realm of plants, the mapping of quantitative trait loci (QTL) initially occurred in bi-parental crosses, but this approach had limitations in terms of allelic diversity and genomic resolution [@borevitz2003]. The genome-wide association studies (GWAS) method addresses various constraints associated with conventional gene mapping. It achieves greater resolution, often down to the gene level, and uses samples from populations that have been extensively studied in the past to enable the association of commonly occurring genetic variations with phenotypic differences. The introduction of high-density single-nucleotide polymorphism (SNP) typing has facilitated comprehensive whole-genome scans, identifying small haplotype blocks significantly correlated with variations in quantitative traits [@brachi2011]. In typical GWAS, phenotype and genotype data are collected for a large sample of assembled individuals such as a diversity panel. The genotype data usually consist of genome-wide SNPs identified through resequencing, genotyping-by-sequencing, or array-based genotyping. The genetic markers most associated with the phenotype of interest are found using statistical methods.   

Soybean (G. max (L.) Merr.) originated in China and is a major source of protein for humans and as a high-quality animal feed. Moreover, the presence of important food supplements in soybean, and growing consumption, has resulted in higher demands for soybean production. Soybean was originally domesticated in China, with about 23000 cultivars in Asia, and was introduced into the USA and Brazil [@lopez2010]. Soybean constitutes one of the largest sources of vegetable oil and of animal protein feed in the world [@sugiyama2015]. It has the highest protein content (40–42%) of all other food crops and is second only to groundnut with respect to the oil content (18–22%) among food legumes [@wynstra1986]. Moreover, soybean is used for aquaculture and biofuel, as well as a protein source for the human diet [@masuda2009].

Different GWAS works have been done in soybean. For example, more than 60 markers identified through GWAS have been reported to be associated with maturity [@1, @2], mainly found in the chromosome 16. In addition to that, 68 markers affecting plant height have been identified using GWAS [6, 7, 9, 11, 12], where 19 markers were mapped on chromosome 19. Moreover, a study of 250 soybean accessions did GWAS analyses for grain yield, maturity, plant height, and seed weight over three years [@ravelombola2021]. 


# Methodology
## Experimental settings
The data set in this study comprised a total of 382 genetically diverse soybean accessions, a subset of the USDA Soybean Germplasm Collection, originated from 15 countries, with maturity groups ranging from 3 to 5. The collection was genotyped with the SoySNP50K BeadChip [@song2013], with approximatelly 50000 markers, and the data has been made available by the authors [@song2015] at SoyBase Genetics and Genomics Database (http://soybase.org/snps/download.php). Field trials were conducted in three environments for two years (2020-2021) using a randomized complete block design with two replicates at Portageville, MO and Clarkton, MO. The trait was the soybean response to off-target dicamba. The off-target dicamba damage was measured in a scale ranging from 1 to 5, with 0.5 increments as described by @vieira_diff. After that, best linear unbiased estimates (BLUEs) were calculated to account for experimental design. Across the three testing environments, the frequency of adjusted off-target dicamba damage scores was consistent and normally distributed with over 45% of the observations between scores of 2 and 3 (moderately tolerant) and 39% between 3 and 3.5 (moderately susceptible). About 8% of the observations were either under the score of 2 (highly tolerant) or above the score of 4 (highly susceptible). More details about the data sets can be found in @vieira2022.

## Model
To identify possible candidate genes associated with off-target dicamba, a genome-wide association studies (GWAS) model was fitted using BLINK [@huang2018] through the GAPIT 3 package [@wang2021]. BLINK eliminates the assumption that causal genes are distributed equally across the genome to improve statistical power by using the linkage disequilibrium (LD) method. A total of three components was used for the Principal Component Analysis (PCA).


```{r setup, include=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = FALSE, include = TRUE)

source("http://zzlab.net/GAPIT/GAPIT.library.R")
source("http://zzlab.net/GAPIT/gapit_functions.txt")

pheno <- read.table("Dicamba_Score.txt", header = TRUE)
geno <- read.table("DicambaSoy50K.hmp.txt", header = FALSE)
```


# Results
```{r, warning = F, include = F}
# https://zzlab.net/GAPIT/gapit_help_document.pdf
model <- GAPIT(Y = pheno, G = geno, PCA.total = 3, model = "BLINK", Geno.View.output = T)
```

<!-- 1. What is the phenotypic distribution for the trait across all testing genotypes? Do we see a normal distribution and a good representation of all possible phenotypes? -->
The following histogram shows the absolute frequency of adjusted off-target Dicamba score: 

```{r, fig.width = 6, fig.asp = 0.6}
hist(pheno$Dicamba, xlab = "Adjusted off-target Dicamba score", main = "")
```

The adjusted response seems to be normally distributed. The original measures range from 1 to 5, with increments of 0.5, but the category "5" is not present in this adjusted response variable. In fact, the minimum and maximum adjusted scores are `r min(pheno$Dicamba)` and `r max(pheno$Dicamba)`, respectively.

<!-- 2. What is the molecular marker density? Is it well distributed across all 20 chromosomes? -->
There are 20 chromosomes, and the marker density across chromosomes was checked:

```{r, fig.width = 8, fig.asp = 0.5}
marker_density <- sort(prop.table(table(geno[-1, "V3"])))
plot(marker_density, xlab = "Chromossome", ylab = "Density")
abline(h = 1 / length(marker_density), col = "blue", lty = 2)
```

Although some chromosomes have more density than others, in general, the distribution of markers seems uniform across chromosomes. The dashed blue line represents the density if the distribution was strictly even ($\frac{1}{\# \text{chromosomes}} = \frac{1}{20} = 0.05$).


<!-- 3. What are the significant marker-trait associations identified? Provide the marker name, chromosome, position, p-value, minor allele frequency, and LOD score. -->

Two markers were identified as significant using significance level of $\alpha = 1 \times 10^{-6}$, as showed by the Manhattan plot: 

```{r}
knitr::include_graphics("GAPIT.Association.Manhattan_Geno.BLINK.Dicamba.pdf")
```

One marker is in chromosome 10, and another is in chromosome 19. The marker name, chromosome, position, p-value, minor allele frequency (MAF), and logarithm of odds (LOD) are as follows:
```{r}
sig_markers <- read.csv("GAPIT.Association.Filter_GWAS_results.csv")
sig_markers <- sig_markers[, c("SNP", "Chr", "Pos", "P.value", "MAF")]
sig_markers$LOD <- -log10(sig_markers$P.value)
colnames(sig_markers) <- c("SNP", "Chromosome", "Position", "p-value", "MAF", "LOD")
knitr::kable(sig_markers, align = "c", row.names = F)
```



<!-- 4. How much of the phenotypic variance is explained by the significant markers (R^2)? -->
When looking at phenotypic variance explained ($R^2$), first marker (`ss715605561` on chromosome 10) explains about $15.1\%$ and second marker (`ss715635349` on chromosome 19) explains about $4.5\%$.


<!-- 5. Based on the QQ Plot, is the model well-fitted for this dataset?  -->
A QQ-plot was produced to check goodness-of-fitness for the GWAS model:

```{r}
knitr::include_graphics("GAPIT.Association.QQ.BLINK.Dicamba.pdf")
```

The model seems to be well-fitted because the non-significant SNPs lie in the diagonal line, whereas two significant markers deviate from the diagonal line in the upper right section of the plot and most likely are associated with the trait. The grey shaded area represents the $95\%$ confidence interval for the QQ-plot under the null hypothesis of no association between the SNP and the trait.   

<!-- 6. Based on the PCA, is the genetic diversity present in this population well distributed? -->
Based on the following PCA plot, the population seems to be well distributed across the three-dimensional PCA space:

```{r}
knitr::include_graphics("GAPIT.Genotype.PCA_3D.pdf")
```


# References